dataset/get-metadata_test.go

```diff
@@ -75,78 +73,96 @@ func TestUnitHandlers(t *testing.T) {
 			Version: 1,
 		}
 
+		datasetCollectionItem := zebedee.CollectionItem{
+			ID:           mockDatasetDetails.ID,
+			State:        "inProgress",
+			LastEditedBy: "an-user",
+		}
+
 		mockCollection := zebedee.Collection{
-			ID: "test-collection",
+			ID: mockCollectionId,
 			Datasets: []zebedee.CollectionItem{
+
 				{
-					ID:    "foo",
-					State: "inProgress",
+					ID:           "foo",
+					State:        "reviewed",
+					LastEditedBy: "other-user",
 				},
+				datasetCollectionItem,
 			},
 		}
 
+		responseHeaders := dataset.ResponseHeaders{ETag: "version-etag"}
+
 		mockZebedeeClient := &ZebedeeClientMock{
 			GetCollectionFunc: func(ctx context.Context, userAccessToken, collectionID string) (c zebedeeclient.Collection, err error) {
-				return mockCollection, nil
+				if collectionID == mockCollectionId {
+					return mockCollection, nil
+				} else {
+					return c, errors.New("Collection not found")
```

Comment:
Error messages need to be all lowercase for clarity in logs:
```suggestion
					return c, errors.New("collection not found")
```